Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from dask.distributed import Client

client = Client()  # or Client("<scheduler URL>") for remote cluster
client
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/deploy/spec.py:308[0m, in [0;36mSpecCluster._start[0;34m(self)[0m
[1;32m    307[0m     [38;5;28mcls[39m [38;5;241m=[39m import_term([38;5;28mcls[39m)
[0;32m--> 308[0m [38;5;28mself[39m[38;5;241m.[39mscheduler [38;5;241m=[39m [38;5;28;43mcls[39;49m[43m([49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mscheduler_spec[49m[38;5;241;43m.[39;49m[43mget[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43moptions[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[43m{[49m[43m}[49m[43m)[49m[43m)[49m
[1;32m    309[0m [38;5;28mself[39m[38;5;241m.[39mscheduler [38;5;241m=[39m [38;5;28;01mawait[39;00m [38;5;28mself[39m[38;5;241m.[39mscheduler

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/scheduler.py:3503[0m, in [0;36mScheduler.__init__[0;34m(self, loop, delete_interval, synchronize_worker_interval, services, service_kwargs, allowed_failures, extensions, validate, scheduler_file, security, worker_ttl, idle_timeout, interface, host, port, protocol, dashboard_address, dashboard, http_prefix, preload, preload_argv, plugins, contact_address, transition_counter_max, jupyter, **kwargs)[0m
[1;32m   3502[0m         http_server_modules[38;5;241m.[39mappend([38;5;124m"[39m[38;5;124mdistributed.http.scheduler.missing_bokeh[39m[38;5;124m"[39m)
[0;32m-> 3503[0m routes [38;5;241m=[39m [43mget_handlers[49m[43m([49m
[1;32m   3504[0m [43m    [49m[43mserver[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[43m,[49m[43m [49m[43mmodules[49m[38;5;241;43m=[39;49m[43mhttp_server_modules[49m[43m,[49m[43m [49m[43mprefix[49m[38;5;241;43m=[39;49m[43mhttp_prefix[49m
[1;32m   3505[0m [43m[49m[43m)[49m
[1;32m   3506[0m [38;5;28mself[39m[38;5;241m.[39mstart_http_server(routes, dashboard_address, default_port[38;5;241m=[39m[38;5;241m8787[39m)

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/http/utils.py:40[0m, in [0;36mget_handlers[0;34m(server, modules, prefix)[0m
[1;32m     39[0m [38;5;28;01mfor[39;00m module_name [38;5;129;01min[39;00m modules:
[0;32m---> 40[0m     module [38;5;241m=[39m [43mimportlib[49m[38;5;241;43m.[39;49m[43mimport_module[49m[43m([49m[43mmodule_name[49m[43m)[49m
[1;32m     41[0m     _routes[38;5;241m.[39mextend(module[38;5;241m.[39mroutes)

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/importlib/__init__.py:127[0m, in [0;36mimport_module[0;34m(name, package)[0m
[1;32m    126[0m         level [38;5;241m+[39m[38;5;241m=[39m [38;5;241m1[39m
[0;32m--> 127[0m [38;5;28;01mreturn[39;00m [43m_bootstrap[49m[38;5;241;43m.[39;49m[43m_gcd_import[49m[43m([49m[43mname[49m[43m[[49m[43mlevel[49m[43m:[49m[43m][49m[43m,[49m[43m [49m[43mpackage[49m[43m,[49m[43m [49m[43mlevel[49m[43m)[49m

File [0;32m<frozen importlib._bootstrap>:1014[0m, in [0;36m_gcd_import[0;34m(name, package, level)[0m

File [0;32m<frozen importlib._bootstrap>:991[0m, in [0;36m_find_and_load[0;34m(name, import_)[0m

File [0;32m<frozen importlib._bootstrap>:975[0m, in [0;36m_find_and_load_unlocked[0;34m(name, import_)[0m

File [0;32m<frozen importlib._bootstrap>:671[0m, in [0;36m_load_unlocked[0;34m(spec)[0m

File [0;32m<frozen importlib._bootstrap_external>:843[0m, in [0;36mexec_module[0;34m(self, module)[0m

File [0;32m<frozen importlib._bootstrap>:219[0m, in [0;36m_call_with_frames_removed[0;34m(f, *args, **kwds)[0m

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/http/scheduler/missing_bokeh.py:3[0m
[1;32m      1[0m [38;5;28;01mfrom[39;00m [38;5;21;01m__future__[39;00m [38;5;28;01mimport[39;00m annotations
[0;32m----> 3[0m [38;5;28;01mfrom[39;00m [38;5;21;01mdistributed[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdashboard[39;00m[38;5;21;01m.[39;00m[38;5;21;01mcore[39;00m [38;5;28;01mimport[39;00m _min_bokeh_version
[1;32m      4[0m [38;5;28;01mfrom[39;00m [38;5;21;01mdistributed[39;00m[38;5;21;01m.[39;00m[38;5;21;01mhttp[39;00m[38;5;21;01m.[39;00m[38;5;21;01mutils[39;00m [38;5;28;01mimport[39;00m RequestHandler, redirect

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/dashboard/core.py:6[0m
[1;32m      4[0m [38;5;28;01mimport[39;00m [38;5;21;01mwarnings[39;00m
[0;32m----> 6[0m [38;5;28;01mfrom[39;00m [38;5;21;01mbokeh[39;00m[38;5;21;01m.[39;00m[38;5;21;01mapplication[39;00m [38;5;28;01mimport[39;00m Application
[1;32m      7[0m [38;5;28;01mfrom[39;00m [38;5;21;01mbokeh[39;00m[38;5;21;01m.[39;00m[38;5;21;01mapplication[39;00m[38;5;21;01m.[39;00m[38;5;21;01mhandlers[39;00m[38;5;21;01m.[39;00m[38;5;21;01mfunction[39;00m [38;5;28;01mimport[39;00m FunctionHandler

[0;31mModuleNotFoundError[0m: No module named 'bokeh'

The above exception was the direct cause of the following exception:

[0;31mRuntimeError[0m                              Traceback (most recent call last)
Cell [0;32mIn [16], line 3[0m
[1;32m      1[0m [38;5;28;01mfrom[39;00m [38;5;21;01mdask[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdistributed[39;00m [38;5;28;01mimport[39;00m Client
[0;32m----> 3[0m client [38;5;241m=[39m [43mClient[49m[43m([49m[43m)[49m  [38;5;66;03m# or Client("<scheduler URL>") for remote cluster[39;00m
[1;32m      4[0m client

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/client.py:982[0m, in [0;36mClient.__init__[0;34m(self, address, loop, timeout, set_as_default, scheduler_file, security, asynchronous, name, heartbeat_interval, serializers, deserializers, extensions, direct_to_workers, connection_limit, **kwargs)[0m
[1;32m    979[0m preload_argv [38;5;241m=[39m dask[38;5;241m.[39mconfig[38;5;241m.[39mget([38;5;124m"[39m[38;5;124mdistributed.client.preload-argv[39m[38;5;124m"[39m)
[1;32m    980[0m [38;5;28mself[39m[38;5;241m.[39mpreloads [38;5;241m=[39m preloading[38;5;241m.[39mprocess_preloads([38;5;28mself[39m, preload, preload_argv)
[0;32m--> 982[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mstart[49m[43m([49m[43mtimeout[49m[38;5;241;43m=[39;49m[43mtimeout[49m[43m)[49m
[1;32m    983[0m Client[38;5;241m.[39m_instances[38;5;241m.[39madd([38;5;28mself[39m)
[1;32m    985[0m [38;5;28;01mfrom[39;00m [38;5;21;01mdistributed[39;00m[38;5;21;01m.[39;00m[38;5;21;01mrecreate_tasks[39;00m [38;5;28;01mimport[39;00m ReplayTaskClient

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/client.py:1172[0m, in [0;36mClient.start[0;34m(self, **kwargs)[0m
[1;32m   1170[0m     [38;5;28mself[39m[38;5;241m.[39m_started [38;5;241m=[39m asyncio[38;5;241m.[39mensure_future([38;5;28mself[39m[38;5;241m.[39m_start([38;5;241m*[39m[38;5;241m*[39mkwargs))
[1;32m   1171[0m [38;5;28;01melse[39;00m:
[0;32m-> 1172[0m     [43msync[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mloop[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_start[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/utils.py:406[0m, in [0;36msync[0;34m(loop, func, callback_timeout, *args, **kwargs)[0m
[1;32m    404[0m [38;5;28;01mif[39;00m error:
[1;32m    405[0m     typ, exc, tb [38;5;241m=[39m error
[0;32m--> 406[0m     [38;5;28;01mraise[39;00m exc[38;5;241m.[39mwith_traceback(tb)
[1;32m    407[0m [38;5;28;01melse[39;00m:
[1;32m    408[0m     [38;5;28;01mreturn[39;00m result

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/utils.py:379[0m, in [0;36msync.<locals>.f[0;34m()[0m
[1;32m    377[0m         future [38;5;241m=[39m asyncio[38;5;241m.[39mwait_for(future, callback_timeout)
[1;32m    378[0m     future [38;5;241m=[39m asyncio[38;5;241m.[39mensure_future(future)
[0;32m--> 379[0m     result [38;5;241m=[39m [38;5;28;01myield[39;00m future
[1;32m    380[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m:
[1;32m    381[0m     error [38;5;241m=[39m sys[38;5;241m.[39mexc_info()

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tornado/gen.py:762[0m, in [0;36mRunner.run[0;34m(self)[0m
[1;32m    759[0m exc_info [38;5;241m=[39m [38;5;28;01mNone[39;00m
[1;32m    761[0m [38;5;28;01mtry[39;00m:
[0;32m--> 762[0m     value [38;5;241m=[39m [43mfuture[49m[38;5;241;43m.[39;49m[43mresult[49m[43m([49m[43m)[49m
[1;32m    763[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m:
[1;32m    764[0m     exc_info [38;5;241m=[39m sys[38;5;241m.[39mexc_info()

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/client.py:1238[0m, in [0;36mClient._start[0;34m(self, timeout, **kwargs)[0m
[1;32m   1235[0m [38;5;28;01melif[39;00m [38;5;28mself[39m[38;5;241m.[39m_start_arg [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m   1236[0m     [38;5;28;01mfrom[39;00m [38;5;21;01mdistributed[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdeploy[39;00m [38;5;28;01mimport[39;00m LocalCluster
[0;32m-> 1238[0m     [38;5;28mself[39m[38;5;241m.[39mcluster [38;5;241m=[39m [38;5;28;01mawait[39;00m LocalCluster(
[1;32m   1239[0m         loop[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39mloop,
[1;32m   1240[0m         asynchronous[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39m_asynchronous,
[1;32m   1241[0m         [38;5;241m*[39m[38;5;241m*[39m[38;5;28mself[39m[38;5;241m.[39m_startup_kwargs,
[1;32m   1242[0m     )
[1;32m   1243[0m     address [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mcluster[38;5;241m.[39mscheduler_address
[1;32m   1245[0m [38;5;28mself[39m[38;5;241m.[39m_gather_semaphore [38;5;241m=[39m asyncio[38;5;241m.[39mSemaphore([38;5;241m5[39m)

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/deploy/spec.py:398[0m, in [0;36mSpecCluster.__await__.<locals>._[0;34m()[0m
[1;32m    396[0m [38;5;28;01masync[39;00m [38;5;28;01mdef[39;00m [38;5;21m_[39m():
[1;32m    397[0m     [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mstatus [38;5;241m==[39m Status[38;5;241m.[39mcreated:
[0;32m--> 398[0m         [38;5;28;01mawait[39;00m [38;5;28mself[39m[38;5;241m.[39m_start()
[1;32m    399[0m     [38;5;28;01mawait[39;00m [38;5;28mself[39m[38;5;241m.[39mscheduler
[1;32m    400[0m     [38;5;28;01mawait[39;00m [38;5;28mself[39m[38;5;241m.[39m_correct_state()

File [0;32m/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/distributed/deploy/spec.py:319[0m, in [0;36mSpecCluster._start[0;34m(self)[0m
[1;32m    317[0m [38;5;28mself[39m[38;5;241m.[39mstatus [38;5;241m=[39m Status[38;5;241m.[39mfailed
[1;32m    318[0m [38;5;28;01mawait[39;00m [38;5;28mself[39m[38;5;241m.[39m_close()
[0;32m--> 319[0m [38;5;28;01mraise[39;00m [38;5;167;01mRuntimeError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mCluster failed to start: [39m[38;5;132;01m{[39;00me[38;5;132;01m}[39;00m[38;5;124m"[39m) [38;5;28;01mfrom[39;00m [38;5;21;01me[39;00m

[0;31mRuntimeError[0m: Cluster failed to start: No module named 'bokeh'
RuntimeError: Cluster failed to start: No module named 'bokeh'

